fsspec/exceptions.py

```diff
@@ -0,0 +1,19 @@
+"""
+fsspec user-defined exception classes
+"""
+
+
+class FSBaseException(Exception):
+    """
+    Base exception for fsspec user-defined exceptions
+    """
+
+    ...
+
+
+class FSTimeoutError(FSBaseException):
```

Comment:
Should derive from the asyncio or concurrent.futures timeout. Note that there is also a `builtins.TimeoutError`

Reply:
Yes, right, and the BaseException can be ignored if we derive from asyncio.TimeoutError.